packages/ui-react/src/containers/AccountModal/forms/ChooseOffer.tsx

```diff
@@ -57,29 +84,54 @@ const ChooseOffer = () => {
 
   const visibleOffers = values.selectedOfferType === 'tvod' ? mediaOffers : isSwitch ? switchSubscriptionOffers : subscriptionOffers;
 
+  const offersRef = useRef<Offer[]>([]);
+
   useEffect(() => {
     if (isLoading || !visibleOffers.length) return;
 
     const offerId = visibleOffers[visibleOffers.length - 1]?.offerId;
 
-    setValue('selectedOfferId', offerId);
-  }, [visibleOffers, values.selectedOfferType, setValue, isLoading]);
+    setValues((currValues) => {
+      if (currValues.selectedOfferId) return currValues;
+
+      return { ...currValues, selectedOfferId: offerId };
+    });
+  }, [visibleOffers, values.selectedOfferType, setValues, isLoading]);
 
   useEffect(() => {
     if (isLoading || !defaultOfferType) return;
 
     setValue('selectedOfferType', defaultOfferType);
   }, [isLoading, defaultOfferType, setValue]);
 
+  useLayoutEffect(() => {
+    offersRef.current = defaultOfferType === 'tvod' ? mediaOffers : isSwitch ? switchSubscriptionOffers : subscriptionOffers;
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [defaultOfferType]);
+
   // loading state
-  if (isLoading) {
+  if (isLoading || redirectUrlRef.current) {
     return (
       <div style={{ height: 300 }}>
         <LoadingOverlay inline />
       </div>
     );
   }
 
+  if (isAccessBridgeEnabled) {
+    return (
+      <ChoosePlanForm
```

Comment:
@mirovladimitrovski do you see the way to fully switch to `ChoosePlanForm` and remove `ChooseOfferForm` in the future?

Reply:
It'll certainly be preferrable to have a unified flow, so yes.